### PR TITLE
Simplify the WhatsApp tests

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by rowanseymour** [comment]
@norkans7 can you provide an explanation of what is changing here... I'm a bit lost

**Comment by norkans7** [comment]
We now use the `utils.whatsapp` to create the templates but we have a `ChannelType.get_api_templates` which do the requests to the API

So I moved the tests for the API in the channel type module 
and those related to the local creation and update of the DB objects in `utils.whatsapp.tests.py`

File: temba/channels/types/dialog360/tests.py [hunk 1]
```diff
@@ -99,227 +101,118 @@ def test_tempalates(self, mock_get):
...
-                "status": "APPROVED",
-               "namespace": "xxxxxxxx_xxxx_xxxx_xxxx_xxxxxxxxxxxx",
-                "rejected_reason": "NONE",
-                "category": "ISSUE_RESOLUTION"
-              }
-                    ]
-                }
-                """,
-            ),
+            MockResponse(200, '{"waba_templates": ["foo", "bar"]}'),
```
Comment (norkans7) [inline temba/channels/types/dialog360/tests.py:107]: Here we mock the API response not need of the complete API response that is tested in `utils.whatsapp.tests`

File: temba/channels/types/dialog360/tests.py [hunk 2]
```diff
@@ -99,227 +101,118 @@ def test_tempalates(self, mock_get):
...
-        mock_get.assert_called_with(
-            "https://example.com/whatsapp/v1/configs/templates",
-            headers={
-                "D360-Api-Key": channel.config[Channel.CONFIG_AUTH_TOKEN],
-                "Content-Type": "application/json",
+        mock_get_api_templates.assert_called_with(channel)
+        self.assertEqual(3, mock_get_api_templates.call_count)
+        update_local_templates_mock.assert_called_once_with(channel, [{"name": "hello"}])
+
+    def test_message_templates_and_logs_views(self):
```
Comment (norkans7) [inline temba/channels/types/dialog360/tests.py:186]: test the views in their own functions

File: temba/utils/whatsapp/tasks.py [hunk 3]
```diff
@@ -92,6 +92,65 @@ def _calculate_variable_count(content):
     return count
 
 
+def update_local_templates(channel, templates_data):
```
Comment (norkans7) [inline temba/utils/whatsapp/tasks.py:95]: This is the method that creates and update the DB templates

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**